examples/src/main/python/SparkSQLExample.py

```diff
@@ -0,0 +1,208 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+# $example on:init_session$
+from pyspark.sql import SparkSession
+# $example off:init_session$
+
+# $example on:schema_infer$
+# spark is an existing SparkSession.
+from pyspark.sql import Row
+# $example off:schema_infer$
+
+# $example on:schema_spec$
+# Import SparkSession and data types
+from pyspark.sql.types import *
+# $example off:schema_spec$
+
+"""
+A simple example demonstrating Spark SQL.
+Run with:
+  ./bin/spark-submit examples/src/main/python/SparkSQLExample.py
+"""
+
+if __name__ == "__main__":
+    # $example on:init_session$
+    spark = SparkSession\
+        .builder\
+        .appName("PythonSQL")\
+        .config("spark.some.config.option", "some-value")\
+        .getOrCreate()
+    # $example off:init_session$
+
+    # $example on:create_df$
+    # spark is an existing SparkSession
+    df = spark.read.json("examples/src/main/resources/people.json")
+
+    # Displays the content of the DataFrame to stdout
+    df.show()
+    # age  name
+    # null Michael
+    # 30   Andy
+    # 19   Justin
+    # $example off:create_df$
+
+    # $example on:untyped_ops$
+    # spark, df are from the previous example
+    # Print the schema in a tree format
+    df.printSchema()
+    # root
+    # |-- age: long (nullable = true)
+    # |-- name: string (nullable = true)
+
+    # Select only the "name" column
+    df.select("name").show()
+    # name
+    # Michael
+    # Andy
+    # Justin
+
+    # Select everybody, but increment the age by 1
+    df.select(df['name'], df['age'] + 1).show()
```

Comment:
@rxin @cloud-fan Shall we deprecate, or at least not recommend, using `df['...']` to reference columns since this may lead to potential ambiguity in self-join cases? I'm thinking about replacing it with `col('...')` in all Python example code.